impl/src/main/java/org/ehcache/config/builders/ExpiryPolicyBuilder.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.config.builders;
+
+import org.ehcache.ValueSupplier;
+import org.ehcache.config.Builder;
+import org.ehcache.expiry.ExpiryPolicy;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * Builder and utilities for getting predefined {@link ExpiryPolicy} instances.
+ */
+public final class ExpiryPolicyBuilder<K, V> implements Builder<ExpiryPolicy<K, V>>{
+
+  /**
+   * Get an {@link ExpiryPolicy} instance for a non expiring (ie. "eternal") cache.
+   *
+   * @return the no expiry instance
+   */
+  public static ExpiryPolicy<Object, Object> noExpiration() {
+    return ExpiryPolicy.NO_EXPIRY;
+  }
+
+  /**
+   * Get a time-to-live (TTL) {@link ExpiryPolicy} instance for the given {@link Duration}.
+   *
+   * @param timeToLive the TTL duration
+   * @return a TTL expiry
+   */
+  public static ExpiryPolicy<Object, Object> timeToLiveExpiration(Duration timeToLive) {
+    Objects.requireNonNull(timeToLive, "TTL duration cannot be null");
+    if (timeToLive.isNegative()) {
+      throw new IllegalArgumentException("TTL duration cannot be negative");
+    }
+    return new TimeToLiveExpiryPolicy(timeToLive);
+  }
+
+  /**
+   * Get a time-to-idle (TTI) {@link ExpiryPolicy} instance for the given {@link Duration}.
+   *
+   * @param timeToIdle the TTI duration
+   * @return a TTI expiry
+   */
+  public static ExpiryPolicy<Object, Object> timeToIdleExpiration(Duration timeToIdle) {
```

Comment:
Shouldn't these return `ExpiryPolicy<K, V>` (inferred by the context)

Reply:
We always consume an `ExpiryPolicy<? super K, ? super V>` so I do not thing the type inference here would buy us much. And since the values are not used, it is correct to say it is an `ExpiryPolicy<Object, Object>`

Reply:
Fine with me.